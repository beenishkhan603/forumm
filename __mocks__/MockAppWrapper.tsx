import { AuthContext } from '@libs/useAuth'
import { MockedProvider, MockedResponse } from '@apollo/client/testing'
import { ProfileInfo } from '@type/Hooks/useAuth/Auth.type'
import ALL_MOCKS from './graphql/mocks'

export const defaultMockProfile: ProfileInfo = {
  company: '448 Studio',
  email: 'test@email.com',
  fullName: 'John Doe',
  university: '448 Studio',
  profileImageUrl: '/test.png',
  userId: '123',
  groups: [],
}

const MockAppWrapper = ({
  children,
  mocks,
  mockProfile = defaultMockProfile,
}: {
  mockProfile?: Partial<ProfileInfo>
  children: any
  mocks?: readonly MockedResponse<Record<string, any>>[] | undefined
}) => {
  const Layout = children?.type?.Layout ?? null

  const GQL_MOCKS: MockedResponse[] = ALL_MOCKS

  if (mocks) mocks.forEach((m) => GQL_MOCKS.push(m))
  return (
    <MockedProvider
      mocks={GQL_MOCKS}
      defaultOptions={{
        query: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-first',
        },
        watchQuery: {
          errorPolicy: 'all',
          fetchPolicy: 'cache-and-network',
        },
        mutate: {
          errorPolicy: 'all',
        },
      }}
    >
      <AuthContext.Provider
        value={{
          profile: mockProfile
            ? { ...defaultMockProfile, ...mockProfile }
            : defaultMockProfile,
          setProfile: () => {},
        }}
      >
        <div id="modal-root" />
        {Layout ? <Layout>{children}</Layout> : children}
      </AuthContext.Provider>
    </MockedProvider>
  )
}

export default MockAppWrapper
