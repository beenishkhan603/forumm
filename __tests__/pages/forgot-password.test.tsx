import { render } from '@testing-library/react'
import ForgotPassword from 'pages/forgot-password'
import MockAppWrapper from '__mocks__/MockAppWrapper'

it('renders forgot password page unchanged', () => {
  const { container } = render(
    <MockAppWrapper>
      <ForgotPassword />
    </MockAppWrapper>
  )
  expect(container).toMatchSnapshot()
})
