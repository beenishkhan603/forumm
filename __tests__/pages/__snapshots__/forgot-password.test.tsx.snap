// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders forgot password page unchanged 1`] = `
<div>
  <div
    id="modal-root"
  />
  <div
    class="flex justify-center h-screen items-center"
    style="background-image: url(/img.jpg); background-size: 100% 100%; background-repeat: no-repeat; background-position: center; background-color: rgb(255, 255, 255);"
  >
    <div
      class="flex w-full justify-center h-screen items-center"
      style="color: rgb(24, 26, 32);"
    >
      <div
        class="flex w-full justify-center items-center max-w-7xl flex-grow overflow-visible"
        style="color: rgb(24, 26, 32);"
      >
        <div
          class="w-4/5 sm:w-1/2 my-16 relative"
          style="color: rgb(24, 26, 32);"
        >
          <div
            class="flex flex-col items-center space-y-2"
            style="color: rgb(24, 26, 32);"
          >
            <div
              class="text-center text-2xl cursor-pointer"
              style="color: rgb(24, 26, 32);"
            >
              <div
                class="flex justify-center items-center mb-12"
              >
                <svg
                  class="block w-12 mr-2"
                />
                <svg
                  class="h-6 w-24"
                />
              </div>
              <p>
                Forgot your password?
              </p>
            </div>
          </div>
          <div
            style="color: rgb(24, 26, 32);"
          >
            <form
              class=""
            >
              <label
                class="flex flex-col my-4 "
              >
                <span
                  class="absolute text-sm mb-2 ml-2 z-10 pl-2 transition-all  !backdrop-blur-md"
                  style="transform: translateY(-50%);"
                >
                  <div
                    class="flex flex-col"
                    style="color: rgb(24, 26, 32);"
                  >
                    <span
                      class="z-10  after:content-['*'] after:mx-2 after:text-red-500"
                    >
                      Email
                    </span>
                  </div>
                </span>
                <input
                  autocomplete="Email"
                  class="rounded-2xl transition-all bg-transparent border  rounded px-4 py-4 text-sm outline-none placeholder-gray-500"
                  id="Email"
                  placeholder="your@mail.com"
                  required=""
                  type="email"
                  value=""
                />
              </label>
              <button
                class="
      bg-forumm-blue-2 text-white relative rounded-full  px-6 py-4 text-base h-8 flex items-center hover:animate-heartbeat justify-center my-4 w-full"
                style="background-color: rgb(55, 99, 233);"
                type="submit"
              >
                <span
                  class="relative z-10"
                  style="color: white;"
                >
                  Send
                </span>
              </button>
            </form>
          </div>
          <div
            class="text-sm flex space-x-2 justify-center mb-2 mt-4"
            style="color: rgb(24, 26, 32);"
          >
            <span
              class=""
            >
              Don't have an Account?
            </span>
            <a
              class="text-link hover:underline"
              href="/create-account"
            >
              Sign up
            </a>
          </div>
          <div
            class="text-sm flex space-x-2 justify-center mb-8"
            style="color: rgb(24, 26, 32);"
          >
            <span
              class=""
            >
              Already registered?
            </span>
            <a
              class="text-link hover:underline"
              href="/login"
            >
              Log in
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
